staging/src/k8s.io/api/flowcontrol/v1alpha1/types.go

```diff
@@ -0,0 +1,340 @@
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+}
+
+// PolicyRulesWithSubjects prescribes a test that applies to a request to an apiserver. The test considers the subject
+// making the request, the verb being requested, and the resource to be acted upon.
+type PolicyRulesWithSubjects struct {
+	// `subjects` is the list of normal user, serviceaccount, or group that this rule cares about.
+	// There must be at least one subject for this field, or it will be rejected by the api validation.
+	// Required.
+	Subjects []Subject `json:"subjects" protobuf:"bytes,1,opt,name=subjects"`
+	// `resourceRules` is the target verb, resource or the subresource the rule cares about. APIGroups, Resources, etc.
+	// +optional
+	ResourceRules []ResourcePolicyRule `json:"resourceRules,omitempty" protobuf:"bytes,2,opt,name=resourceRules"`
+	// `nonResourceRules` is the target verb, non-resource urls, etc.
+	// +optional
+	NonResourceRules []NonResourcePolicyRule `json:"nonResourceRules,omitempty" protobuf:"bytes,3,opt,name=nonResourceRules"`
+}
+
+// ResourcePolicyRule holds information that describes a policy rule for resources, but does not contain information
+// about who the rule applies to or which namespace the rule applies to.
+type ResourcePolicyRule struct {
+	// `verbs` is a list of Verbs that apply to ALL the Resource and Kinds contained in this rule.
+	// "*" represents all verbs.
+	// Required.
+	Verbs []string `json:"verbs" protobuf:"bytes,1,rep,name=verbs"`
+	// `apiGroups` is the name of the APIGroup that contains the resources.  If multiple API groups are specified, any
+	// action requested against one of the enumerated resources in any API group will be allowed.
+	// Note that exactly one of this slice and nonResourceURLs is empty.
+	// "*" represents all api-groups.
+	// Required.
+	APIGroups []string `json:"apiGroups" protobuf:"bytes,2,rep,name=apiGroups"`
+	// `resources` is a list of resources this rule applies to.  ResourceAll represents all resources.
+	// Note that exactly one of this slice and nonResourceURLs is empty.
+	// "*" represents all resource urls.
+	// Required.
+	Resources []string `json:"resources" protobuf:"bytes,3,rep,name=resources"`
+}
+
+// NonResourcePolicyRule holds information that describes a policy rule for non-resources, but does not contain information
+// about who the rule applies to or which namespace the rule applies to.
+type NonResourcePolicyRule struct {
+	// `verbs` is a list of Verbs that apply to ALL the Resource and Kinds contained in this rule.
+	// "*" represents all verbs.
+	// Required.
+	Verbs []string `json:"verbs" protobuf:"bytes,1,rep,name=verbs"`
+	// `nonResourceURLs` is a set of partial urls that a user should have access to.  *s are allowed, but only as the full, final step in the path
+	// Rules can either apply to API resources (such as "pods" or "secrets") or non-resource URL paths (such as "/api"),  but not both.
+	// "*" represents all non-resource urls.
+	// Required.
```

Comment:
updated to:

> 	// `nonResourceURLs` is a set of partial urls that a user should have access to.
	// "*" represents all non-resource urls.
	// Required.

Reply:
Since the more extensive commentary has been removed, it is no longer clear what is meant by "partial URL".  But any reasonable interpretation of that phrase is inconsistent with the implementation we have so far.

Reply:
Instead of "partial URLs", maybe "URL prefixes"? An example would make this clearer.

Reply:
I am OK with "illegal" iff validation rejects "illegal" things.

Reply:
And let's not forget that we have not implemented what's documented here.